### DISCUSSION TITLE
Where is the compiled llama directory

### DISCUSSION THREAD

**Discussion by NEMESYS43** [withcatai/node-llama-cpp#134] [General]
I have successfully compiled Llama for my setup previously. I am using ROCm 5.6 with a Radeon 7900xtx. When using the Cuda build guide in the docs I can't seem to get it to successfully compile. Can't I just replace the llama binary that this library uses with the one I already have compiled? has anyone found where it is stored in this case? I have been digging around and have not been able to find any clues.

**Comment by giladgd** [comment]
@NEMESYS43 This project compiles `llama.cpp` with custom code to bridge it properly to nodejs, so you can't provide your own binary for this.
You can follow the instructions in [this comment](https://github.com/withcatai/node-llama-cpp/issues/84#issuecomment-1807234486) to compile it for AMD ROC, but note that `llama.cpp`'s support for OpenCL is deprecated and can be removed at any time.
